pilot/pkg/model/authorization.go

```diff
@@ -191,6 +196,23 @@ func (policy *AuthorizationPolicies) RoleToBindingsForNamespace(ns string) map[s
 	return rolesAndBindings.RoleNameToBindings
 }
 
+// RoleForNameAndNamespace returns a ServiceRole from this namespace, given the ServiceRole name and namespace.
+// This function always return a non nil struct instance.
+func (policy *AuthorizationPolicies) RoleForNameAndNamespace(roleName, ns string) *rbacproto.ServiceRole {
+	if policy == nil || policy.NamespaceToAuthorizationConfigV2 == nil {
```

Comment:
Not with it current usage, since it has been checked here. 
https://github.com/istio/istio/blob/72330246e863731fccc67dd8ff77ce749636a6cd/pilot/pkg/networking/plugin/authz/rbac_v2.go#L56

However, caller are not guaranteed to have non-nil `AuthorizationPolicies`, so I think we still need it?

Reply:
I mean policy is the class object, can it be called with nil value. I'm not too familiar with the way golang support class, though it sounds very strange that you can do smt like nil.RoleForNameAndNamespace, isn't it?

Reply:
But isn't it a pointer to the `AuthorizationPolicies` struct?